packages/react-instantsearch/src/core/createConnector.js

```diff
@@ -105,7 +105,7 @@ export default function createConnector(connectorDesc) {
         props: this.getProps(nextProps),
       });
 
-      if (isWidget) {
+      if (isWidget && !isEqual(nextProps, this.props)) {
```

Comment:
This is the optimization to avoid making a new search if new props are the same as the old props.